src/isolation/filemap.rs

```diff
@@ -43,17 +43,19 @@ impl UhyveFileMap {
 	/// Returns the host_path on the host filesystem given a requested guest_path, if it exists.
 	///
 	/// * `guest_path` - The guest path that is to be looked up in the map.
-	pub fn get_host_path(&mut self, guest_path: &str) -> Option<OsString> {
+	pub fn get_host_path(&self, guest_path: &str) -> Option<OsString> {
+		let try_resolve_path = |path: &Path| {
+			path.to_str()
+				.and_then(|requested_path| self.files.get(requested_path))
+		};
+
 		// TODO: Replace clean-path in favor of Path::normalize_lexically, which has not
 		// been implemented yet. See: https://github.com/rust-lang/libs-team/issues/396
 		let requested_guest_pathbuf = clean(guest_path);
-		let host_path = self
-			.files
-			.get(&requested_guest_pathbuf.display().to_string())
-			.map(OsString::from);
-		trace!("get_host_path (host_path): {host_path:#?}");
-		if host_path.is_some() {
-			host_path
+		if let Some(host_path) = try_resolve_path(&requested_guest_pathbuf) {
```

Comment:
Completely missed `try_resolve_path` - I didn't really like using `is_some()` at the time but had time constraints. Good one.